### PR TITLE
W-037259: [EDA] Create custom settings to prevent deletion

### PR DESCRIPTION
**Pull request by kdy618**
# Critical Changes

# Changes
We have updated the Hierarchy Custom Settings to include 9 new checkbox fields. These checkboxes, designed for internal-use only, provide the scaffolding for a forthcoming deletion-prevention feature; they currently do not provide any functionality. We strongly advise that EDA Admins do not manage Hierarchy Custom Settings directly and instead use the EDA Settings UI.

# Issues Closed

# New Metadata

# Deleted Metadata

# Testing Notes
Please see description column on this spreadsheet. https://salesforce.quip.com/oU2qAs876VyW#CQFACAEqoyM

### CODE REVIEW

**Comment by robertbwatson** [comment]
@kdy618 @juliesnow  Just wanted to add a comment here that I don't think this should go under Critical Changes. We might also want to consider language that makes it clear that these new settings do absolutely nothing at this time and are placeholders for future work.

**Comment by xinnanlisf** [comment]
@dashattuck it looks like the robot test is still failing after several attempts. Could you take a look at it please? Thank you!

**Comment by kdy618** [comment]
@robertbwatson  @xinnanlisf  ty so much! Moved the description to below critical change.

**Comment by kdy618** [comment]
@juliesnow , could you kindly take a look if these new checkboxes should be under changes?

**Comment by juliesnow** [comment]
> @kdy618 @juliesnow are these descriptions still a work in progress? I ask because I don't think the plan is to prevent deletion unless there are certain EDA child records present. If we just leave it as preventing deletion, people may think that we will block deletion outright. Maybe I'm not on the same page, but I didn't that was the case.

@ryanguest - Not sure if we've changed them since you made this comment, so please take another look. The descriptions indicate which child records are involved, and I've matched the descriptions for existing classes to the descriptions for those classes in the Apex docs, so I believe them to be accurate.

File: src/objects/Hierarchy_Settings__c.object [hunk 1]
```diff
@@ -355,6 +355,96 @@
         <type>Text</type>
         <unique>false</unique>
     </fields>
+    <fields>
+        <fullName>Prevent_Account_Deletion__c</fullName>
+        <defaultValue>false</defaultValue>
+        <description>Prevent an Account from being deleted if it has any related records (for example, Affiliation, Program Enrollment, or Course Connection records).</description>
```
Comment (robertbwatson) [inline src/objects/Hierarchy_Settings__c.object]: @kdy618 @lifewithryan @xinnanlisf  Is this ready for QA or do we need to make changes here?

(Did we decide earlier on "Reserved for future use."? I'm not sure what we ultimately decided!)

**Comment by dgamboni** [comment]
@robertbwatson Do you know if this closes any issues? It looks like the settings don't actually do anything so I'm guessing it doesn't close any of the open issues related to deletion at this time.

**Comment by xinnanlisf** [comment]
@dgamboni no it doesn't have a github issue I believe since it is an internal item. @kdy618 correct me if I am wrong.

**Comment by lifewithryan** [comment]
> @robertbwatson Do you know if this closes any issues? It looks like the settings don't actually do anything so I'm guessing it doesn't close any of the open issues related to deletion at this time.

@dgamboni -- correct this doesn't really close any issues. It's opening the door for us to being handling the data integrity items.

**Comment by kdy618** [comment]
@xinnanlisf, yes you are correct. This is an internal item. @dgamboni

File: src/objects/Hierarchy_Settings__c.object (hunk 1 above)
Comment (kdy618) [reply src/objects/Hierarchy_Settings__c.object]: @robertbwatson , yep we will use "Reserved for future use." This is now back to dev review. :)

**Comment by dgamboni** [comment]
> @dgamboni no it doesn't have a github issue I believe since it is an internal item. @kdy618 correct me if I am wrong.

@xinnanlisf @lifewithryan @jacebryan I don't think this is a truly internal task, though, because there potential for customers to see these changes if they go to Setup > Custom Settings (as opposed to, say, a cci task for our qa_org flows, which it seems customers would never be exposed to.) So, I'm thinking maybe it would technically need an issue? (Though I'm not advocating for more 'paperwork', as it were, and am 100% fine if we don't have a related issue -- I'm just trying to grok the rationale.)

**Comment by dgamboni** [comment]
Tested / Passes QE. @spinkelman Will be merging this PR.

**Comment by spinkelman** [comment]
Merging per @ dgamboni request.

**Pull request merged**